kedro/extras/datasets/tensorflow/tensorflow_model_dataset.py

```diff
@@ -151,6 +151,10 @@ def _load(self) -> tf.keras.Model:
     def _save(self, data: tf.keras.Model) -> None:
         save_path = get_filepath_str(self._get_save_path(), self._protocol)
 
+        # Make sure all intermediate directories are created.
+        save_dir = Path(save_path).parent
+        save_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
I'm surprised line 115 isn't enough here. 🤔  Do you know more about why that is?

Reply:
Digging into `fsspec`, it looks like this is a peculiarity with how that library works. Specifically, setting `auto_mkdir=True` will automatically create the directory, but not immediately. The directory will be created lazily when the `_open()` method gets called. However, AFAIK, `TensorflowModelDataset._save()` never calls `_open()`. For HDF5 files, it eventually calls `copy()`, but in `fsspec`, this merely delegates to `shutil.copy()` without calling `_open()`. Thus, `fsspec` fails to create the parent directory automatically, although, arguably, it should.

If you think that this issue would be better fixed in `fsspec`, or that the code here should include a `TODO` explaining the situation, I would be amenable to making those alterations.

Incidentally, it looks like this issue is resolved in the [current `master` branch](https://github.com/intake/filesystem_spec/blob/efcba4503de0b9f9ba3b12fa59e48285ffe8b382/fsspec/implementations/local.py#L83) of `fsspec`. Maybe the proper resolution here is just to update `fsspec`?

Reply:
Oh that's interesting, thank you for the very thorough research!! 
Is that a released `fsspec` version? We have a ticket to further relax the `fsspec` boundaries to include 0.8.x (0.7.x was a breaking change for us so that bump was done on `develop`). Do you have time to look into whether that would also fix this issue, and if so we can merge a PR with that update instead (against `develop`)?

Reply:
A cursory glance at the `fsspec` code indicates that the issue is fixed in `0.8.0`. It seems, then, that merging the existing PR would indeed fix this issue also. If you agree, I will go and test locally to make sure that the issue is indeed resolved.

Reply:
:+1:

Should I be handling the second PR too or will someone else do that?

Reply:
If you'd like to take that on that would be super helpful! 😊  If not no worries someone in the team can do it.

Reply:
I can definitely take a crack at this. I might not get to it until tomorrow though.

Reply:
I'm definitely up for merging this into master as-is for `0.16.6`. Then we can open another PR agaisnt `develop`. I think you can create another PR and reference the discussion in this PR. No need to create another issue for it. WDYT?

Reply:
I agree